readme.md
## About

This package emerged through the need of a custom state manager for 
[IBMs Carbon Design Data table component](https://carbondesignsystem.com/components/data-table/usage/).
The default behavior of the component (sorting, filtering, searching) can only be overwritten by 
implementing an own Custom State manager.

Read more about it here: https://github.com/carbon-design-system/carbon/issues/6373#issuecomment-747612917

After implementing this custom state manager, I realized that I need the same code in many projects, so this 
package was born.

The initial code was taken from the example implementation by IBM: https://github.com/carbon-design-system/carbon/issues/6373#issuecomment-747612917

## Features

The package was written with a focus on displaying time series data. 

### Time 

You can supply a column named "time" and an optional [Day.js format string](https://day.js.org/docs/en/display/format). 
The time column will then be formatted according to this format string (default is YYYY/MM/DD HH:mm). Additionaly if you supply
a title prop, the Table will show a description in the form `from <start date> till <end date>`.

## Requirements

**Requires Carbon components v10, migration to v11 is coming soon.**

```
"@carbon/icons-react": "10.x",
"carbon-components": "10.x",
"carbon-components-react": "7.x",
"react": ">=16",
"react-dom": ">=16"
```

## Install

```sh
yarn add carbon-data-table-state-manager
```

or

```sh
npm i carbon-data-table-state-manager --save
```

## Usage

### Example

```
import { CarbonDataTableStateManager } from "carbon-data-table-state-manager";
import { TABLE_SORT_DIRECTION } from "carbon-data-table-state-manager/misc";

const columns = [
  {
    title: "Time",
    id: "time",
    sortCycle: "tri-states-from-ascending",
  },
  {
    title: "Temperature",
    small: `in °C`,
    id: "temperature",
    sortCycle: "tri-states-from-ascending",
  },
  {
    title: "Humidity",
    small: `in %`,
    id: "humidity",
    sortCycle: "tri-states-from-ascending",
  },
];

const rows = [
  {
    time: 1614927420,
    temperature: -0.7,
    humidity: 93,
  },
  {
    time: 2021-03-05T02:57:00.000Z,
    temperature: 1.2,
    humidity: 77,
  },
];

<CarbonDataTableStateManager
  columns={columns}
  rows={rows}
  start={0}
  pageSize={10}
  pageSizes={[10, 25, 50, 100, 200, 500]}
  sortInfo={{
    columnId: "timeParsed",
    direction: TABLE_SORT_DIRECTION.ASC,
  }}
  size="short"
/>
```

### Storybook

See the [Storybook](https://daveiano.github.io/carbon-data-table-state-manager) for more examples.

### Props

<table>
  <tr>
    <th>Prop</th>
    <th>Type</th>
    <th>Default</th>
    <th>Description</th>
  </tr>
  <tr>
    <td>collator?</td>
    <td>Intl.Collator</td>
    <td>new Intl.Collator()</td>
    <td>An Intl.Collator object for string comparison.</td>
  </tr>
  <tr>
    <td>columns</td>
    <td>

```
{
  id: string | number;
  title: string;
  sortCycle?: string;
  small?: string;
  tooltip?: string;
}[]
```

</td>
    <td></td>
    <td>The table headers / column definitions</td>
  </tr>
  <tr>
    <td>hasSelection?</td>
    <td>boolean</td>
    <td>false</td>
    <td>Should the table allow for selection?</td>
  </tr>
  <tr>
    <td>id?</td>
    <td>string</td>
    <td></td>
    <td></td>
  </tr>
  <tr>
    <td>pageSize</td>
    <td>number</td>
    <td>5</td>
    <td>Default page size.</td>
  </tr>
  <tr>
    <td>rows</td>
    <td>

```
{
  [key: string]: any;
  selected: boolean;
  time?: string | number;
}[]
```

</td>
    <td></td>
    <td>The actual table rows.</td>
  </tr>
  <tr>
    <td>size?</td>
    <td>"compact" | "lg" | "md" | "normal" | "short" | "sm" | "tall" | "xl" | "xs"</td>
    <td>normal</td>
    <td>Table cell size.</td>
  </tr>
  <tr>
    <td>sortInfo</td>
    <td>

```
{
  columnId: string;
  direction: "ASC" | "DESC" | "NONE";
}
```

</td>
    <td></td>
    <td>Default sorting state</td>
  </tr>
   <tr>
    <td>start</td>
    <td>number</td>
    <td>0</td>
    <td>Offset</td>
  </tr>
   <tr>
    <td>zebra?</td>
    <td>boolean</td>
    <td></td>
    <td>Use zebra styling for table rows</td>
  </tr>
   <tr>
    <td>title?</td>
    <td>string</td>
    <td></td>
    <td>Table title</td>
  </tr>
   <tr>
    <td>dateFormat?</td>
    <td>string</td>
    <td></td>
    <td>

[Day.js format string](https://day.js.org/docs/en/display/format) for formatting the `time` column

</td>
  </tr>
   <tr>
    <td>pageSizes?</td>
    <td>number[]</td>
    <td></td>
    <td>Array of available page sizes for the Pagination.</td>
  </tr>
   <tr>
    <td>emptyCell?</td>
    <td>string</td>
    <td>"-"</td>
    <td>String to render when column has no data.</td>
  </tr>
</table>

changelog.md
# 0.0.1-alpha.1.0

- Initial working package
- Added rollup

# 0.0.1-alpha2.0

- Added `emptyCell` prop
- Added Storybook

# 0.0.1-beta1.0

- Added various stories to Storybook

# 1.0.0

- Added jest and basic test coverage

# 1.0.1

- Fixed build bug

# 1.0.2

- Added dangerouslySetInnerHTML for rendering `small` in table headers

# 1.0.3

- Changed '–' to '-' in pagination for better compatibility.

# Next

stories/data.ts
import { TABLE_SORT_DIRECTION } from "../src/misc";

export const columns = [
  {
    title: "Time",
    id: "time",
    sortCycle: "tri-states-from-ascending",
  },
  {
    title: "Temperature",
    small: `in &#176;C`,
    id: "temperature",
    sortCycle: "tri-states-from-ascending",
  },
  {
    title: "Felt temperature",
    small: `in °C`,
    tooltip: "Some tooltip.",
    id: "felt_temperature",
    sortCycle: "tri-states-from-ascending",
  },
  {
    title: "Dew point",
    small: `in °C`,
    id: "dew_point",
    sortCycle: "tri-states-from-ascending",
  },
  {
    title: "Pressure",
    small: `in hPa`,
    id: "pressure",
    sortCycle: "tri-states-from-ascending",
  },
  {
    title: "Humidity",
    small: `in %`,
    id: "humidity",
    sortCycle: "tri-states-from-ascending",
  },
  {
    title: "Wind",
    small: `in km/h`,
    id: "wind",
    sortCycle: "tri-states-from-ascending",
  },
  {
    title: "Gust",
    small: `in km/h`,
    id: "gust",
    sortCycle: "tri-states-from-ascending",
  },
  {
    title: "Solar irradiation",
    small: `in w/m²`,
    id: "solar",
    sortCycle: "tri-states-from-ascending",
  },
  {
    title: "UV Index",
    id: "uvi",
    sortCycle: "tri-states-from-ascending",
  },
];

const baseRows = [
  {
    time: 1614913020,
    humidity: 99,
    dew_point: 0.2,
    felt_temperature: 0.4,
    wind: 0,
    gust: 0,
    pressure: 986.1,
    solar: 0,
    uvi: 0,
    id: "2",
  },
  {
    time: 1614920220,
    temperature: -0.2,
    humidity: 96,
    dew_point: -0.7,
    felt_temperature: -0.2,
    wind: 1.4,
    gust: 3.6,
    pressure: 987.6,
    solar: 0,
    uvi: 0,
    id: "3",
  },
  {
    time: 1614927420,
    temperature: -0.7,
    humidity: null,
    dew_point: -1.8,
    felt_temperature: -0.7,
    wind: 1.1,
    gust: 1.8,
    pressure: 989.7,
    solar: 38.1,
    uvi: 0,
    id: "4",
  },
  {
    time: 1614934620,
    temperature: 1.2,
    humidity: 77,
    dew_point: -2.4,
    felt_temperature: 1.2,
    wind: 0,
    gust: 0,
    pressure: 991,
    solar: 154.3,
    uvi: 1,
    id: "5",
  },
  {
    time: 1614941820,
    temperature: 2.4,
    humidity: 58,
    dew_point: -5,
    felt_temperature: 2.4,
    wind: 0.4,
    gust: 1.8,
    pressure: 992.2,
    solar: 106.6,
    uvi: 1,
    id: "6",
  },
  {
    time: 1614949020,
    temperature: 3,
    humidity: 51,
    dew_point: -6.1,
    felt_temperature: 3,
    wind: 0,
    gust: 0,
    pressure: 992.7,
    solar: 293.2,
    uvi: 2,
    id: "7",
  },
  {
    time: 1614956220,
    temperature: 5.8,
    humidity: 39,
    dew_point: -7.1,
    felt_temperature: 5.8,
    wind: 1.1,
    gust: 1.8,
    pressure: 993.6,
    solar: 97.4,
    uvi: 0,
    id: "8",
  },
  {
    time: 1614963420,
    temperature: 0.3,
    humidity: 65,
    dew_point: -5.5,
    felt_temperature: 0.3,
    wind: 1.4,
    gust: 1.8,
    pressure: 995.1,
    solar: 0,
    uvi: 0,
    id: "9",
  },
  {
    time: 1614970620,
    temperature: -2,
    humidity: 76,
    dew_point: -5.7,
    felt_temperature: -2,
    wind: 0,
    gust: 0,
    pressure: 996.6,
    solar: 0,
    uvi: 0,
    id: "10",
  },
  {
    time: 1614977820,
    temperature: -2.1,
    humidity: 72,
    dew_point: -6.5,
    felt_temperature: -2.1,
    wind: 0,
    gust: 0,
    pressure: 998.2,
    solar: 0,
    uvi: 0,
    id: "11",
  },
  {
    time: 1614985020,
    temperature: -2.6,
    humidity: 78,
    dew_point: -5.9,
    felt_temperature: -2.6,
    wind: 0,
    gust: 0,
    pressure: 998.4,
    solar: 0,
    uvi: 0,
    id: "12",
  },
  {
    time: 1614985140,
    temperature: -2.6,
    humidity: 74,
    dew_point: -6.6,
    felt_temperature: -2.6,
    wind: 0,
    gust: 0,
    pressure: 998.5,
    solar: 0,
    uvi: 0,
    _id: "DOeJvDciAjUstzet",
    timeParsed: "2021-03-05T22:59:00.000Z",
    id: "13",
  },
  {
    time: 1614985200,
    temperature: -2.6,
    humidity: 74,
    dew_point: -6.6,
    felt_temperature: -2.6,
    wind: 0,
    gust: 0,
    pressure: 998.5,
    solar: 0,
    uvi: 0,
    _id: "31z3hTmqBER2EVen",
    timeParsed: "2021-03-05T23:00:00.000Z",
    id: "14",
  },
  {
    time: 1614992220,
    temperature: -2.8,
    humidity: 78,
    dew_point: -6.1,
    felt_temperature: -2.8,
    wind: 1.4,
    gust: 1.8,
    pressure: 998.8,
    solar: 0,
    uvi: 0,
    _id: "SPJrynCee33V4EBk",
    timeParsed: "2021-03-06T00:57:00.000Z",
    id: "15",
  },
  {
    time: 1614999420,
    temperature: -3.2,
    humidity: 78,
    dew_point: -6.5,
    felt_temperature: -3.2,
    wind: 0,
    gust: 0,
    pressure: 998.7,
    solar: 0,
    uvi: 0,
    _id: "yLjjS7EMDj54bZkm",
    timeParsed: "2021-03-06T02:57:00.000Z",
    id: "16",
  },
  {
    time: 1615006620,
    temperature: -3.5,
    humidity: 82,
    dew_point: -6.2,
    felt_temperature: -3.5,
    wind: 0,
    gust: 0,
    pressure: 999.6,
    solar: 0,
    uvi: 0,
    _id: "jHom67gi2d0qUUDl",
    timeParsed: "2021-03-06T04:57:00.000Z",
    id: "17",
  },
  {
    time: 1615013820,
    temperature: -1,
    humidity: 72,
    dew_point: -5.4,
    felt_temperature: -3.4,
    wind: 6.5,
    gust: 7.2,
    pressure: 1000.5,
    solar: 46.3,
    uvi: 0,
    _id: "wYqHnOqxdIGGevfd",
    timeParsed: "2021-03-06T06:57:00.000Z",
    id: "18",
  },
  {
    time: 1615021020,
    temperature: 0.7,
    humidity: 67,
    dew_point: -4.8,
    felt_temperature: -1.7,
    wind: 7.2,
    gust: 11.2,
    pressure: 1000.6,
    solar: 278.4,
    uvi: 2,
    _id: "EjYExQY8VgY4kwO7",
    timeParsed: "2021-03-06T08:57:00.000Z",
    id: "19",
  },
  {
    time: 1615028220,
    temperature: 3.3,
    humidity: 56,
    dew_point: -4.6,
    felt_temperature: 0.2,
    wind: 11.9,
    gust: 18.4,
    pressure: 999.6,
    solar: 444.1,
    uvi: 4,
    _id: "SmP4NfHpmfB8Z5Aj",
    timeParsed: "2021-03-06T10:57:00.000Z",
    id: "20",
  },
  {
    time: 1615035420,
    temperature: 4,
    humidity: 55,
    dew_point: -4.2,
    felt_temperature: 0.2,
    wind: 16.9,
    gust: 20.2,
    pressure: 998.8,
    solar: 392.5,
    uvi: 3,
    _id: "dQEqtinrfwZNCusb",
    timeParsed: "2021-03-06T12:57:00.000Z",
    id: "21",
  },
  {
    time: 1615042620,
    temperature: 4.1,
    humidity: 57,
    dew_point: -3.7,
    felt_temperature: 2.2,
    wind: 7.6,
    gust: 9.4,
    pressure: 997.9,
    solar: 81.4,
    uvi: 0,
    _id: "wQFV5jaOw576UulQ",
    timeParsed: "2021-03-06T14:57:00.000Z",
    id: "22",
  },
  {
    time: 1615049820,
    temperature: 2.4,
    humidity: 63,
    dew_point: -3.9,
    felt_temperature: 2.4,
    wind: 4.7,
    gust: 5.4,
    pressure: 997.9,
    solar: 0,
    uvi: 0,
    _id: "Zu05gdvcfmqZGu6m",
    timeParsed: "2021-03-06T16:57:00.000Z",
    id: "23",
  },
  {
    time: 1615057020,
    temperature: 0.9,
    humidity: 76,
    dew_point: -2.9,
    felt_temperature: -1.3,
    wind: 6.8,
    gust: 7.2,
    pressure: 997.2,
    solar: 0,
    uvi: 0,
    _id: "iI0hU9HZo063QUWv",
    timeParsed: "2021-03-06T18:57:00.000Z",
    id: "24",
  },
  {
    time: 1615064220,
    temperature: 1.2,
    humidity: 76,
    dew_point: -2.6,
    felt_temperature: -0.4,
    wind: 5.4,
    gust: 5.4,
    pressure: 996.6,
    solar: 0,
    uvi: 0,
    _id: "4MxHhSZefnx7ZSvJ",
    timeParsed: "2021-03-06T20:57:00.000Z",
    id: "25",
  },
  {
    time: 1615071420,
    temperature: 1.5,
    humidity: 77,
    dew_point: -2.1,
    felt_temperature: -0.7,
    wind: 7.2,
    gust: 7.2,
    pressure: 996,
    solar: 0,
    uvi: 0,
    _id: "z456RZghexNq7RzF",
    timeParsed: "2021-03-06T22:57:00.000Z",
    id: "26",
  },
  {
    time: 1615071540,
    temperature: 1.5,
    humidity: 77,
    dew_point: -2.1,
    felt_temperature: -0.5,
    wind: 6.5,
    gust: 7.2,
    pressure: 995.8,
    solar: 0,
    uvi: 0,
    _id: "GLfx2pjMIYhN1Jza",
    timeParsed: "2021-03-06T22:59:00.000Z",
    id: "27",
  },
  {
    time: 1615071600,
    temperature: 1.5,
    humidity: 77,
    dew_point: -2.1,
    felt_temperature: -0.5,
    wind: 6.5,
    gust: 7.2,
    pressure: 995.8,
    solar: 0,
    uvi: 0,
    _id: "RFMMLmbGuTJiQgSI",
    timeParsed: "2021-03-06T23:00:00.000Z",
    id: "28",
  },
  {
    time: 1615078620,
    temperature: 1.4,
    humidity: 78,
    dew_point: -2,
    felt_temperature: -1.1,
    wind: 8.3,
    gust: 9.4,
    pressure: 995.1,
    solar: 0,
    uvi: 0,
    _id: "VAItNISfHpyMQgDK",
    timeParsed: "2021-03-07T00:57:00.000Z",
    id: "29",
  },
  {
    time: 1615085820,
    temperature: 0.9,
    humidity: 77,
    dew_point: -2.7,
    felt_temperature: -2.7,
    wind: 11.9,
    gust: 14.8,
    pressure: 993.6,
    solar: 0,
    uvi: 0,
    _id: "v5pHjohCMXMnI8sJ",
    timeParsed: "2021-03-07T02:57:00.000Z",
    id: "30",
  },
  {
    time: 1615093020,
    temperature: 0.9,
    humidity: 72,
    dew_point: -3.6,
    felt_temperature: -2.8,
    wind: 12.2,
    gust: 14.8,
    pressure: 992.9,
    solar: 0,
    uvi: 0,
    _id: "uY9ORD7Ufyqg5KZd",
    timeParsed: "2021-03-07T04:57:00.000Z",
    id: "31",
  },
  {
    time: 1615100220,
    temperature: 1.1,
    humidity: 71,
    dew_point: -3.6,
    felt_temperature: -3,
    wind: 14.8,
    gust: 16.6,
    pressure: 992.6,
    solar: 36.8,
    uvi: 0,
    _id: "NnohdA3QGb5ifxgp",
    timeParsed: "2021-03-07T06:57:00.000Z",
    id: "32",
  },
  {
    time: 1615107420,
    temperature: 1.3,
    humidity: 74,
    dew_point: -2.8,
    felt_temperature: -1.7,
    wind: 10.1,
    gust: 13,
    pressure: 992.1,
    solar: 94,
    uvi: 0,
    _id: "jlIdUX5HyGJHCa1T",
    timeParsed: "2021-03-07T08:57:00.000Z",
    id: "33",
  },
  {
    time: 1615114620,
    temperature: 2.2,
    humidity: 74,
    dew_point: -2,
    felt_temperature: -1.2,
    wind: 12.2,
    gust: 16.6,
    pressure: 990.8,
    solar: 78.9,
    uvi: 0,
    _id: "ynzPAybQlyoWssGI",
    timeParsed: "2021-03-07T10:57:00.000Z",
    id: "34",
  },
  {
    time: 1615121820,
    temperature: 1.8,
    humidity: 71,
    dew_point: -2.9,
    felt_temperature: -1.8,
    wind: 13.3,
    gust: 16.6,
    pressure: 989.4,
    solar: 55,
    uvi: 0,
    _id: "at8YU2EzbhV5yk8u",
    timeParsed: "2021-03-07T12:57:00.000Z",
    id: "35",
  },
  {
    time: 1615129020,
    temperature: 2.7,
    humidity: 67,
    dew_point: -2.9,
    felt_temperature: -0.3,
    wind: 10.8,
    gust: 13,
    pressure: 987.9,
    solar: 84,
    uvi: 0,
    _id: "97r4wVhrnhVNb3NJ",
    timeParsed: "2021-03-07T14:57:00.000Z",
    id: "36",
  },
  {
    time: 1615136220,
    temperature: 1.1,
    humidity: 68,
    dew_point: -4.2,
    felt_temperature: -1.2,
    wind: 7.2,
    gust: 11.2,
    pressure: 987.5,
    solar: 0,
    uvi: 0,
    _id: "BmKToEuwPtW1surb",
    timeParsed: "2021-03-07T16:57:00.000Z",
    id: "37",
  },
  {
    time: 1615143420,
    temperature: 0.8,
    humidity: 67,
    dew_point: -4.7,
    felt_temperature: -2.7,
    wind: 11.5,
    gust: 13,
    pressure: 987.2,
    solar: 0,
    uvi: 0,
    _id: "pqVQSYCiTv0QX7ph",
    timeParsed: "2021-03-07T18:57:00.000Z",
    id: "38",
  },
  {
    time: 1615150620,
    temperature: 0.6,
    humidity: 71,
    dew_point: -4.1,
    felt_temperature: -2.9,
    wind: 11.2,
    gust: 13,
    pressure: 987.4,
    solar: 0,
    uvi: 0,
    _id: "FYWzl1fbqlHxKiat",
    timeParsed: "2021-03-07T20:57:00.000Z",
    id: "39",
  },
  {
    time: 1615157820,
    temperature: 0.3,
    humidity: 77,
    dew_point: -3.3,
    felt_temperature: -3.8,
    wind: 13.7,
    gust: 16.6,
    pressure: 986.4,
    solar: 0,
    uvi: 0,
    _id: "fUG7hpMlbfUyiIFb",
    timeParsed: "2021-03-07T22:57:00.000Z",
    id: "40",
  },
  {
    time: 1615157940,
    temperature: 0.3,
    humidity: 78,
    dew_point: -3.1,
    felt_temperature: -3.3,
    wind: 11.5,
    gust: 14.8,
    pressure: 986.7,
    solar: 0,
    uvi: 0,
    _id: "epqaLeLNCvmfkFVy",
    timeParsed: "2021-03-07T22:59:00.000Z",
    id: "41",
  },
  {
    time: 1615158000,
    temperature: 0.3,
    humidity: 78,
    dew_point: -3.1,
    felt_temperature: -3.3,
    wind: 11.5,
    gust: 14.8,
    pressure: 986.7,
    solar: 0,
    uvi: 0,
    _id: "Z3IVmQTpquhMLUF4",
    timeParsed: "2021-03-07T23:00:00.000Z",
    id: "42",
  },
  {
    time: 1615165020,
    temperature: -0.8,
    humidity: 96,
    dew_point: -1.3,
    felt_temperature: -4.6,
    wind: 11.2,
    gust: 13,
    pressure: 986.7,
    solar: 0,
    uvi: 0,
    _id: "2JWKBcIunYP4NHir",
    timeParsed: "2021-03-08T00:57:00.000Z",
    id: "43",
  },
  {
    time: 1615172220,
    temperature: -0.7,
    humidity: 98,
    dew_point: -1,
    felt_temperature: -2.9,
    wind: 6.1,
    gust: 7.2,
    pressure: 986.9,
    solar: 0,
    uvi: 0,
    _id: "3XwHvK92lHjHFI5F",
    timeParsed: "2021-03-08T02:57:00.000Z",
    id: "44",
  },
  {
    time: 1615179420,
    temperature: -0.5,
    humidity: 99,
    dew_point: -0.7,
    felt_temperature: -0.5,
    wind: 0,
    gust: 0,
    pressure: 987.8,
    solar: 0,
    uvi: 0,
    _id: "G2atJKjKAUbZxouF",
    timeParsed: "2021-03-08T04:57:00.000Z",
    id: "45",
  },
  {
    time: 1615186620,
    temperature: -0.2,
    humidity: 99,
    dew_point: -0.4,
    felt_temperature: -0.2,
    wind: 0,
    gust: 0,
    pressure: 988.9,
    solar: 45.5,
    uvi: 0,
    _id: "qE8PsWDfsum90yLd",
    timeParsed: "2021-03-08T06:57:00.000Z",
    id: "46",
  },
  {
    time: 1615193820,
    temperature: 1.2,
    humidity: 80,
    dew_point: -1.9,
    felt_temperature: 1.2,
    wind: 2.5,
    gust: 5.4,
    pressure: 990,
    solar: 247.6,
    uvi: 2,
    _id: "88tHiTkn8LlsluAZ",
    timeParsed: "2021-03-08T08:57:00.000Z",
    id: "47",
  },
  {
    time: 1615201020,
    temperature: 3.5,
    humidity: 56,
    dew_point: -4.5,
    felt_temperature: 3.5,
    wind: 0.7,
    gust: 1.8,
    pressure: 990.1,
    solar: 511.3,
    uvi: 4,
    _id: "5id48SZ8myxO1i1R",
    timeParsed: "2021-03-08T10:57:00.000Z",
    id: "48",
  },
  {
    time: 1615208220,
    temperature: 5.4,
    humidity: 45,
    dew_point: -5.6,
    felt_temperature: 4.5,
    wind: 5,
    gust: 5.4,
    pressure: 989.9,
    solar: 384.7,
    uvi: 3,
    _id: "DzxNdM5JuAKAqdI9",
    timeParsed: "2021-03-08T12:57:00.000Z",
    id: "49",
  },
];

export const rows = baseRows.map((item) => ({
  selected: false,
  ...item,
}));

export const rowsSelected = baseRows.map((row, j) => ({
  ...row,
  selected: j < 3 || j === 8,
}));

export const sortInfo = {
  columnId: "time",
  direction: TABLE_SORT_DIRECTION.ASC,
};
